### PR TITLE
Add check to filter listener from classes

### LINKED ISSUE #40551: JetStreamListener is listed under classes
## Description
The JetStreamListener is listed under classes in https://lib.ballerina.io/ballerinax/nats/latest 
Ideally, this should be under `Listeners`
Screenshot 2023-05-30 at 10 25 28

### PR DESCRIPTION
**Pull request by NipunaMadhushan**
## Purpose
Filter Listener classes from classes with a better approach.

Fixes #40551

## Approach
We cannot use `listener` as a qualifier in a class definition. Therefore the previous method to filter listener classes was to check the name of the class whether it is equal to `Listener`. But it will even filter other classes with name `Listener` as listeners and wouldn't filter listeners with other names.

These changes will check the necessary methods which should be included in a class to be a listener. The following methods should be included in a class to be a listener.
1. `attach`
2. `detach`
3. `'start`
4. `immediateStop`
5. `gracefulStop`

This approach will check for these methods to identify whether it is a listener or not.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java [hunk 1]
```diff
@@ -485,13 +485,38 @@ private static BClass getClassModel(ClassDefinitionNode classDefinitionNode, Sem
...
 
+    private static boolean isListenerModel(List<Function> classFunctions) {
+        boolean isStartIncluded = false;
+        boolean isAttachIncluded = false;
+        boolean isDetachIncluded = false;
+        boolean isGracefulStopIncluded = false;
+        boolean isImmediateStopIncluded = false;
+
+        for (Function function : classFunctions) {
+            if (function.name.equals("'start")) {
```
Comment (keizer619) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java:503]: @NipunaMadhushan Shall we move these into constants?

File: misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java [hunk 2]
```diff
@@ -485,13 +485,38 @@ private static BClass getClassModel(ClassDefinitionNode classDefinitionNode, Sem
...
+                isStartIncluded = true;
```
Comment (keizer619) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java:504]: @NipunaMadhushan We can return true here and just keep `if` conditions and can avoid line 516

File: misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java [hunk 3]
```diff
@@ -485,13 +485,38 @@ private static BClass getClassModel(ClassDefinitionNode classDefinitionNode, Sem
...
+            } else if (function.name.equals("attach")) {
+                isAttachIncluded = true;
+            } else if (function.name.equals("detach")) {
+                isDetachIncluded = true;
+            } else if (function.name.equals("gracefulStop")) {
+                isGracefulStopIncluded = true;
+            } else if (function.name.equals("immediateStop")) {
```
Comment (keizer619) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java:511]: @NipunaMadhushan Also just wondering is it enough to check the name or do we need to validate the signature too

**Review by sm1990 — DISMISSED** [review]

**Review by anuruddhal — APPROVED** [review]

File: misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java (hunk 1 above)
Comment (NipunaMadhushan) [reply misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java]: @keizer619 sure. Addressed the suggestions.

**Review by keizer619 — APPROVED** [review]

**Pull request merged**